### PR TITLE
Needs view page

### PR DESCRIPTION
**Pull request by shmily40686**
-Create page for views
-Link to /needs from Homepage

### CODE REVIEW

File: src/components/needLeadViewPage/needsViewTable.tsx [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+            />
+            <SearchIcon className={styles.searchIcon} />
+          </Container>
+        </Box>
+        <Box className={styles.outerTable}>
+          <div className={styles.tableCustomBorder}></div>
+          <table className={styles.table}>
+            <thead>
+              <tr>
+                <th className={styles.header}>
```
Comment (ComeTiss) [inline src/components/needLeadViewPage/needsViewTable.tsx:209]: Optional: You coud define the columns header in an array before, then do a .map(à on it to render the <th> elements

File: src/components/needLeadViewPage/needsViewTable.tsx [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+import { GET_NEEDS } from "../../service/apollo/queries";
+import Need from "../../service/models/need.model";
+
+type Props = {
+  page: number;
+};
+
+function NeedsViewTable() {
+  const [page, setPage] = useState(1);
+  const [totalPages, setTotalPage] = useState(1);
```
Comment (ComeTiss) [inline src/components/needLeadViewPage/needsViewTable.tsx:18]: why not merging both states together ? Seems like a lot of hooks here

File: src/components/needLeadViewPage/needsViewTable.tsx [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+      request: {
+        options: {
+          page: page,
+          limit: 10
+        },
+        filters: {}
+      }
+    },
+    fetchPolicy: "cache-and-network"
+  });
```
Comment (ComeTiss) [inline src/components/needLeadViewPage/needsViewTable.tsx:131]: I would have put those queries in the higher component (Need page component)

File: src/components/needLeadViewPage/needsViewTable.tsx [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+        needs.push({
+          urgencyLevel: "",
+          itemName: "",
+          createdAt: "",
+          specifications: "",
+          quantity: "",
+          budget: "",
+          id: `spacer-need-${Math.random()}`
+        });
+      }
```
Comment (ComeTiss) [inline src/components/needLeadViewPage/needsViewTable.tsx]: that can be put into a function (as you are already using this code before)

File: src/components/needLeadViewPage/needsViewTable.tsx [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+      setNeeds(totalNeeds);
+      setTotalPage(1);
+    } else {
+      const totalNeeds = filteredNeeds?.getAllNeeds?.needs || [];
+      const needs = needsData?.getAllNeeds?.needs || [];
+      let space;
+      if (needs.length !== 10) {
```
Comment (ComeTiss) [inline src/components/needLeadViewPage/needsViewTable.tsx]: needs.lenght < 10 ? And why are you filling up the table with empty rows?

**Review by ComeTiss — APPROVED** [review]

File: src/components/needLeadViewPage/needsViewTable.tsx (hunk 5 above)
Comment (shmily40686) [reply src/components/needLeadViewPage/needsViewTable.tsx]: Removed the logic for that, I was using it to correctly space the items but I was able to fix it with CSS.

**Pull request merged**